presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedSet.java

```diff
@@ -130,9 +134,29 @@ public void testBigintSimpleTypedSet()
         }
     }
 
+    @Test
+    public void testMemoryExceeded()
```

Comment:
In addition, I would add one test to one of the functions to test the end-to-end behavior.

For example,

`assertInvalidFunction("array_distinct(ARRAY ['" + VERY_LONG_STRING + "']), EXCEEDED_FUNCTION_MEMORY_LIMIT)`